### DISCUSSION TITLE
Adding repository to devcontainer.json does not give access

### DISCUSSION THREAD

**Discussion by mwisnicki** [community/community#107471] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

I wanted to push to another repository from existing workspace.

Following [these instructions](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces#setting-additional-repository-permissions) I've modified `devcontainer.json` as follows:

```json
{
    "image": "mcr.microsoft.com/devcontainers/universal:2",
    "customizations": {
        "codespaces": {
            "repositories": {
                "mwisnicki/something-private": {
                    "permissions": "write-all"
                }
            }
        }
    }
}
```

And rebuilt the container.

However I didn't get a prompt for access as described [here](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces#authorizing-requested-permissions).

And pushing fails:

```
[info] Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by mwisnicki** [comment]
Please note that I've only changed `devcontainer.json` on disk without committing to original repo because I don't have push access to original repo.

**Comment by Olayiwolaaa** [comment]
restart with this 
``` bash
git clone https://PAT:$YOUR-VARIABLE@github.com/OWNER/REPO`
```

# please mark as answer if it works. Also, read this 
`https://docs.github.com/en/codespaces/troubleshooting/troubleshooting-authentication-to-a-repository#authenticating-to-repositories-that-you-didnt-create-the-codespace-from`

**Reply by mwisnicki** [reply]
Per: https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces#authorizing-requested-permissions
> When you authorize permissions for a repository, GitHub Codespaces will not re-prompt you unless the set of requested permissions has changed for the repository.

Am I misinterpreting it?

Anyway I solved it with ssh keys but looks like github could improve either documentation or better yet behavior.
I expected rebuild to pick up new permissions as per quote above.

**Reply by Olayiwolaaa** [reply, marked as answer]
Oh you're correct. Others call it scope. 
Permission, Scope, i guess their same thing.
It definitely won't prompt that page again unless you update the permission, 
> which is logical (you left out the part where you already got that page prompted)
:warning: can i get my response marked as the answer though, i need the brain badge thing.
